doc/classes/String.xml

```diff
@@ -534,13 +600,30 @@
 			<param index="1" name="base" type="int" default="10" />
 			<param index="2" name="capitalize_hex" type="bool" default="false" />
 			<description>
-				Converts a signed [int] to a string representation of a number.
+				Converts the given [param number] to a string representation, with the given [param base].
+				By default, [param base] is set to decimal ([code]10[/code]). Other common bases in programming include binary ([code]2[/code]), [url=https://en.wikipedia.org/wiki/Octal]octal[/url] ([code]8[/code]), hexadecimal ([code]16[/code]).
+				If [param capitalize_hex] is [code]true[/code], digits higher than 9 are represented in uppercase.
 			</description>
 		</method>
 		<method name="num_scientific" qualifiers="static">
 			<return type="String" />
 			<param index="0" name="number" type="float" />
 			<description>
+				Converts the given [param number] to a string representation, in scientific notation.
+				[codeblocks]
+				[gdscript]
+				var n = -5.2e8
+				print(n)                       # Prints -520000000
+				print(String.NumScientific(n)) # Prints -5.2e+08
+				[/gdscript]
+				[csharp]
+				// This method is not implemented in C#.
+				// Use `string.ToString()` with "e" to achieve similar results.
+				var n = -5.2e8f;
+				GD.Print(n);               // Prints -520000000
+				GD.Print(n.ToString("e")); // Prints -5.2e+08
```

Comment:
This would actually print `-5.200000e+008`. You can specify the number of decimals so if you use `e1` you would get `-5.2e+008`. I think that's as close as you can get, but I'm not sure if it's worth going into that much detail in the documentation.

It may be useful to add a link to https://learn.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings though.